src/extensions/lens-main-extension.ts

```diff
@@ -10,10 +10,14 @@ import type { CatalogEntity } from "../common/catalog";
 import type { IObservableArray } from "mobx";
 import type { MenuRegistration } from "../main/menu/menu-registration";
 import type { TrayMenuRegistration } from "../main/tray/tray-menu-registration";
+import type { ShellEnvModifier } from "../main/shell-session/shell-env-modifier/shell-env-modifier-registration";
+
 export class LensMainExtension extends LensExtension {
   appMenus: MenuRegistration[] = [];
   trayMenus: TrayMenuRegistration[] = [];
 
+  terminalShellEnvModifier?: ShellEnvModifier;
```

Comment:
It would be great if we could get some documentation this. Could be a follow up PR.

Reply:
Is this the place to put the documentation?

Reply:
It is one of the two places to do so. Any doc comments (of the form `/** * */`) above items such as fields, methods, functions, types, ... will be rendered as part of https://api-docs.k8slens.dev/v5.3.4/extensions/api/

The other option is of course guides.

Reply:
None of the other fields are documented here, so I wondered if they're documented elsewhere, like where they're (or the lens main extension is) exported

Reply:
They are not documented at all, which is sort of a problem. I mentioned it because I think we should try to improve the state here but I don't think it should block this PR.